client/core/core.go

```diff
@@ -1573,7 +1575,25 @@ func (c *Core) ReconfigureWallet(appPW []byte, assetID uint32, cfg map[string]st
 	// Reload the wallet with the new settings.
 	wallet, err := c.loadWallet(dbWallet)
 	if err != nil {
-		return newError(walletErr, "error loading wallet for %d -> %s: %v", assetID, unbip(assetID), err)
+		return newError(walletErr, "error loading wallet for %d -> %s: %v",
+			assetID, unbip(assetID), err)
+	}
+
+	newPasswordSet := newWalletPW != nil // INcludes empty but non-nil
+	// If newWalletPW is non-nil, update the wallet's password.
+	if newPasswordSet {
+		// Encrypt password if it's not an empty string
+		encPW := newWalletPW
+		if string(newWalletPW) != "" {
+			encPW, err = crypter.Encrypt(newWalletPW)
+			if err != nil {
+				return newError(encryptionErr, "encryption error: %v", err)
+			}
+		}
+		err = c.setWalletPassword(wallet, newWalletPW, encPW)
```

Comment:
Doesn't `setWalletPassword` unlock the wallet? It should definitely check the password, but this is a change in behavior for a user setting a password on a locked wallet, since the wallet will be unlocked afterwards. There was some previous effort made to ensure the lock state was the same after as it was before. 

In `SetWalletPassword`, this is handled with the `wasUnlocked` variable. You might consider moving that logic into the unexported `setWalletPassword` method.

Reply:
moved